apps/cowswap-frontend/src/modules/injectedWidget/hooks/useAppCodeWithWidgetMetadata.ts

```diff
@@ -0,0 +1,46 @@
+import { useMemo } from 'react'
+
+import { environmentName } from '@cowprotocol/common-utils'
+
+import { AppDataWidget } from 'modules/appData/types'
+
+import { useInjectedWidgetMetaData } from './useInjectedWidgetMetaData'
+
+export interface AppCodeWithWidgetMetadata {
+  appCode: string
+  environment?: string
+  widget?: AppDataWidget
```

Comment:
Could the widget part be independent?

I feel confused looking at this type as this is a partial appData (appCode and env) with a metadata (widget).

Also, it's called appCode and has env in it

Reply:
Used this other name, in case is more clear https://github.com/cowprotocol/cowswap/pull/3371/commits/40143914284429b05db24245afc3f4ec4a559402